src/Kiota.Builder/Validation/OpenApiSchemaComparer.cs

```diff
@@ -27,7 +27,7 @@ public OpenApiSchemaComparer(
     /// <inheritdoc/>
     public bool Equals(OpenApiSchema? x, OpenApiSchema? y)
     {
-        return EqualsInternal(x, y);
+        return x == null && y == null || x != null && y != null && GetHashCode(x) == GetHashCode(y);
```

Comment:
I think there should be a doc comment with a note on the behavior if collisions happen.